enterprise/metrics/src/main/java/org/neo4j/metrics/output/CsvOutput.java

```diff
@@ -59,36 +64,47 @@ public CsvOutput( Config config, MetricRegistry registry, Log logger, KernelCont
     @Override
     public void init()
     {
-        if ( config.get( csvEnabled ) )
+        // Setup CSV reporting
+        File configuredPath = config.get( csvPath );
+        if ( configuredPath == null )
         {
-            // Setup CSV reporting
-            File configuredPath = config.get( csvPath );
-            if ( configuredPath == null )
-            {
-                throw new IllegalArgumentException( csvPath.name() + " configuration is required since " +
-                        csvEnabled.name() + " is enabled" );
-            }
-            outputFile = absoluteFileOrRelativeTo( kernelContext.storeDir(), configuredPath );
-            MetricsSettings.CsvFile csvFile = config.get( MetricsSettings.csvFile );
-            switch ( csvFile )
-            {
-            case single:
-                csvReporter = CsvReporterSingle.forRegistry( registry )
-                        .convertRatesTo( TimeUnit.SECONDS )
-                        .convertDurationsTo( TimeUnit.MILLISECONDS )
-                        .filter( MetricFilter.ALL )
-                        .build( outputFile );
-                break;
-            case split:
-                csvReporter = CsvReporter.forRegistry( registry )
-                        .convertRatesTo( TimeUnit.SECONDS )
-                        .convertDurationsTo( TimeUnit.MILLISECONDS )
-                        .filter( MetricFilter.ALL )
-                        .build( ensureDirectoryExists( outputFile ) );
-                break;
-            default: throw new IllegalArgumentException(
-                    "Unsupported " + MetricsSettings.csvFile.name() + " setting: " + csvFile );
-            }
+            throw new IllegalArgumentException( csvPath.name() + " configuration is required since " +
+                                                csvEnabled.name() + " is enabled" );
+        }
+        outputPath = absoluteFileOrRelativeTo( kernelContext.storeDir(), configuredPath );
+        csvReporter = CsvReporter.forRegistry( registry )
+                .convertRatesTo( TimeUnit.SECONDS )
+                .convertDurationsTo( TimeUnit.MILLISECONDS )
+                .filter( MetricFilter.ALL )
+                .build( ensureDirectoryExists( outputPath ) );
+    }
+
+    @Override
+    public void start()
+    {
+        csvReporter.start( config.get( csvInterval ), TimeUnit.MILLISECONDS );
+        logger.info( "Sending metrics to CSV file at " + outputPath );
+    }
+
+    @Override
+    public void stop() throws IOException
+    {
+        csvReporter.stop();
+    }
+
+    @Override
+    public void shutdown()
+    {
+        csvReporter = null;
+    }
+
+    @Override
+    public void report( SortedMap<String,Gauge> gauges, SortedMap<String,Counter> counters,
+            SortedMap<String,Histogram> histograms, SortedMap<String,Meter> meters, SortedMap<String,Timer> timers )
+    {
+        synchronized ( csvReporter )
```

Comment:
i still can read that as called by one concrete polling thread, and that can be source of confusion - why to synchronize